DOCUMENTATION.md

```diff
@@ -1370,529 +1395,202 @@ The following are example configurations.
+A pseudo-code is as follows:
 
-### 6.1. TLS/SSL
+```text
+begin_time = getCurrentTime()
+current_backoff_millis = INITIAL_BACKOFF_MILLIS
+while (tryConnect(connectionTimeout)) != SUCCESS) {
+    if (getCurrentTime() - begin_time >= CLUSTER_CONNECT_TIMEOUT_MILLIS) {
+        // Give up to connecting to the current cluster and switch to another if exists.
+    }
+    Sleep(current_backoff_millis + UniformRandom(-JITTER * current_backoff_millis, JITTER * current_backoff_millis))
+    current_backoff = Min(current_backoff_millis * MULTIPLIER, MAX_BACKOFF_MILLIS)
+}
+```
 
-One of the offers of Hazelcast is the TLS/SSL protocol which you can use to establish an encrypted communication across your cluster with key stores and trust stores.
+Note that, `tryConnect` above tries to connect to any member that the client knows, and for each connection we
+have a connection timeout; see the [Setting Connection Timeout](#54-setting-connection-timeout) section.
 
-* A Java `keyStore` is a file that includes a private key and a public certificate. The equivalent of a key store is the combination of `key` and `cert` files at the Node.js client side.
-* A Java `trustStore` is a file that includes a list of certificates trusted by your application which is named as  "certificate authority". The equivalent of a trust store is a `ca` file at the Node.js client side.
+# 7. Using Node.js Client with Hazelcast IMDG
 
-You should set `keyStore` and `trustStore` before starting the members. See the next section on setting `keyStore` and `trustStore` on the server side.
+This chapter provides information on how you can use Hazelcast IMDG's data structures in the Node.js client, after giving some basic information including an overview to the client API, operation modes of the client and how it handles the failures.
 
-#### 6.1.1. TLS/SSL for Hazelcast Members
+## 7.1. Node.js Client API Overview
 
-Hazelcast allows you to encrypt socket level communication between Hazelcast members and between Hazelcast clients and members, for end to end encryption. To use it, see the [TLS/SSL for Hazelcast Members section](http://docs.hazelcast.org/docs/latest/manual/html-single/index.html#tls-ssl-for-hazelcast-members).
+Most of the functions in the API return `Promise`. Therefore, you need to be familiar with the concept of promises to use the Node.js client. If not, you can learn about them using various online resources, e.g., the [Promise JS](https://www.promisejs.org/) website.
 
-#### 6.1.2. TLS/SSL for Hazelcast Node.js Clients
+Promises provide a better way of working with callbacks. You can chain asynchronous functions by the `then()` function of promise. Also, you can use `async/await`, if you use Node.js 8 and higher versions.
```

Comment:
nit: remove the first two sentences in this paragraph and move the remaining sentence into the previous paragraph. Everyone knows about promises nowadays.